common/src/stack/command/stack/argument_processors/vm.py

```diff
@@ -0,0 +1,191 @@
+		Raises a CommandError if one or more hosts are not
+		valid virtual machines
+		"""
+
+		hosts = self.getHostnames(args)
+
+		# Use valid_vm to determine if the vm is
+		# defined
+		vm_hosts = [host for host in hosts if self.valid_vm(host)]
+
+		# Need to check if args is empty or not
+		# as getHostnames will return all hosts if args
+		# is empty and in that case we skip the check
+		if args and vm_hosts != hosts:
+			raise CommandError(self, f'One or more hosts are not a valid virtual machine')
+		return vm_hosts
+
+	def get_hypervisor_by_name(self, vm_name):
+		"""
+		Get the hostname of a vm's hypervisor via
+		the vm hostname
+		"""
+
+		vm_id = self.vm_id_by_name(vm_name)
+		result = self.db.select(
+			"""
+			nodes.name
+			FROM nodes
+			INNER JOIN virtual_machines
+			ON nodes.id = virtual_machines.hypervisor_id
+			AND virtual_machines.id = %s
+			""", (vm_id, )
+		)
+		if result:
+			return result[0][0]
+
+	def is_hypervisor(self, hypervisor):
+		"""
+		Determine if the input host is a valid hypervisor
+		via it's appliance.
+		"""
+
+		if not hypervisor:
+			return False
+		attr = self.getHostAttr(hypervisor, 'hypervisor')
+		return str2bool(attr)
+	def vm_info(self, hosts):
+		"""
+		Get virtual specific attributes
+		from the database and return it
+		with the hostnames as keys
+		"""
+
+		vm_info = defaultdict(list)
+
+		# Copy the hosts arg to append extra parameters
+		for row in self.db.select(
+			"""
+			nodes.name,
+			(SELECT name FROM nodes WHERE nodes.id = vm.hypervisor_id)
+			AS hypervisor, vm.memory_size, vm.cpu_cores, vm.vm_delete
+			FROM nodes
+			INNER JOIN virtual_machines vm
+			ON nodes.id = vm.node_id
+			"""):
+			if row[0] in hosts:
+				vm_info[row[0]].extend(row[1:-1])
+
+				# Turn the integer value stored in the database to a
+				# boolean for display
+				vm_info[row[0]].append(bool(row[-1]))
+
+		return vm_info
+
+	def get_all_disks(self, hosts):
+		"""
+		Return a dict of vm disk info ordered by hostname
+		"""
+
+		disks = defaultdict(dict)
+		for row in self.db.select(
+			"""
+			nodes.name, vmd.id, disk_name, disk_type, disk_location, disk_size, image_file_name, mount_disk, vmd.disk_delete
+			FROM virtual_machine_disks vmd
+			INNER JOIN virtual_machines vm
+			ON vmd.virtual_machine_id = vm.id
+			INNER JOIN nodes
+			ON vm.node_id = nodes.id
+			"""):
+				if row[0] in hosts:
+					disks[row[0]][row[1]] = list(row[2:-1])
+
+					# Turn disk deletion value into a bool
+					# for display
+					disks[row[0]][row[1]].append(bool(row[-1]))
+		return disks
+
+	def delete_pending_disk(self, host, disk_name):
+		"""
+		For a given host and disk name, delete it
+		from the database if it's been marked for
+		deletion prior via remove vm storage command
+		"""
+
+		vm_id = self.vm_id_by_name(host)
+		return self.db.execute(
+			"""
+			DELETE FROM virtual_machine_disks
+			WHERE virtual_machine_disks.virtual_machine_id = %s
+			AND virtual_machine_disks.disk_delete
+			IS TRUE
```

Comment:
Just FYI: You normally wouldn't break `IS TRUE` on to its own line, just the same as you wouldn't put `=foo` on its own line. You mainly just break up WHERE clauses at `AND`, `OR`, or `(`.